firmware/broadcom_trx.ksy

```diff
@@ -134,9 +159,44 @@ types:
             pos: ofs_body
             size: len_body
             if: is_present
+          usual_purpose:
+            value: idx
+            enum: usual_purpose
+        enums:
+          usual_purpose:
+            0xFF: unknown
+            0: bootloader
+            1: kernel
+            2: rootfs
+            3: some_header
+
       flags:
+        doc: the the names of these flags will probably be changed
         seq:
-          - id: flags
+          - id: unkn
             type: b1
             repeat: expr
-            repeat-expr: 16
+            repeat-expr: 9
+          - id: is_bootloader # 6
+            -orig-id: TRX_BOOTLOADER
+            type: b1
+          - id: is_uncompressed # 5
+            -orig-id: TRX_UNCOMP_IMAGE
+            type: b1
+            doc: The image is uncompressed
+          - id: is_rom_simulation # 4
+            -orig-id: TRX_ROMSIM_IMAGE
+            type: b1
+          - id: is_microcode # 3
+            -orig-id: TRX_EMBED_UCODE
+            type: b1
+            doc-ref: https://github.com/seemoo-lab/nexmon/blob/297ec9f4e591cee9d7256fa95d644d815222d9e5/utilities/dhdutil/ucode_download.c#L130
```

Comment:
SHA-1 is not collision resistant (and git now has support of SHA256 for local repos because of that, but fetching and pushing is not yet implemented), but truncated hashes are especially. It is improbable that someone has ever needed to create a collision even for a truncated hash for that repo though, but why not just for lulz? You propose to use 7 hex digits (it is by the order of magnitude `16^7 * 2 ≈ 0.5 GH` in a rough assumption that 1 try is 2 sequential hashes), it is feasible not only for a white-box attack, but also for a birthday attack, and for an attack on the second preimage, by the order of magnitude about `2.6 seconds` on Titan RTX [using the hashrate of  `20983.4 MH/s` from](https://gist.github.com/Chick3nman/5d261c5798cf4f3867fe7035ef6dd49f). Of course resistance can be increased by keeping more digits, but anyway, using truncated hashes is a poor practice we shouldn't promote.

Reply:
> it is feasible not only for a white-box attack, but also for a birthday attack, and for an attack on the second preimage

So what? This is not a password hash used for authentication, this is merely a convenient identifier in a link so that the person opening it gets the document snapshot from a specific revision, not from `master` that may have changed a dozen times (or even renamed/deleted completely). The worst thing that can happen is that someone will have nothing better to do than going through the commit identifiers that we use in links, finding collisions on these partial hashes and then somehow injecting a commit with the colliding hash. The only impact of this act is that the commit identifier that we use will become ambiguous and thus Git will refuse to checkout the revision, which will result in GitHub showing a 404 page. If that's the intent, to make our links in KSY specs broken, then sure, it'll be a bit annoying for us and it will take some effort to find the right commit and make the hash a bit longer so that it's no longer ambiguous. But from the perspective of an "attacker", I don't think it's worth it, but sure, prove me wrong and show me some meaningful attack vector that makes good money.

What is this "security" concern really about, @KOLANICH? Do you just want to show how many attack names you know?

Rather than looking for problems in places where there are none, if we really want to increase security and protect our users, we should rather convince @GreyCat to agree on the HTTPS enforcement for *.kaitai.io domains (https://github.com/kaitai-io/kaitai_struct/issues/809), for example. As I see it, that's not just a niche problem, but a real necessity in 2021, unlike the one that we're currently discussing.

Reply:
>But from the perspective of an "attacker", I don't think it's worth it, but sure, prove me wrong and show me some meaningful attack vector that makes good money.

I also think such an attack doesn't worth. But again,  using truncated hashes is a poor practice we should not promote. 

>, if we really want to increase security and protect our users,

we should first get rid of JavaScript on the sites. https enforcement is solved client side, either using HTTPSEverywhere in EASE mode, or `about:config` pref, or both. JS issue is not solved at all by setting the right prefs, the best a user can do is writing an own surrogate script for GreaseMonkey and installing it into the browser.